src/fsipy/automatedPostprocessing/postprocessing_h5py/spectrograms.py

```diff
@@ -0,0 +1,747 @@
+            else f"wss_amplitude_{args.lowcut}_to_100000.csv"
+    else:
+        logging.error(f"ERROR: Invalid value for quantity - {args.quantity}. Please use 'd', 'v', 'p', or 'wss'.")
+        sys.exit(-1)
+
+    return args
+
+
+def read_spectrogram_data(folder: Union[str, Path], mesh_path: Union[str, Path], save_deg: int, stride: int,
+                          start_t: float, end_t: float, n_samples: int, sampling_region: str,
+                          fluid_sampling_domain_id: int, solid_sampling_domain_id: int, fsi_region: list[float],
+                          quantity: str, interface_only: bool, component: str, point_id: int,
+                          fluid_domain_id: Union[int, list[int]], solid_domain_id: Union[int, list[int]],
+                          sampling_method: str = "RandomPoint"):
+    """
+    Read spectrogram data and perform processing steps.
+
+    Args:
+        folder (str or Path): Path to simulation results.
+        mesh_path (str or Path): Path to the mesh file.
+        save_deg (int): Degree of mesh refinement.
+        stride (int): Desired frequency of output data.
+        start_t (float): Start time for data processing.
+        end_t (float): End time for data processing.
+        n_samples (int): Number of samples.
+        sampling_region (str): Region for sampling data ("sphere" or "domain").
+        fluid_sampling_domain_id (int): Domain ID for fluid sampling (used when sampling_region="domain").
+        solid_sampling_domain_id (int): Domain ID for solid sampling (used when sampling_region="domain").
+        fsi_region (list): x, y, and z coordinates of sphere center and radius of the sphere (used when
+            sampling_region="sphere").
+        quantity (str): Quantity to postprocess.
+        interface_only (bool): Whether to include only interface ID's.
+        component (str): Component of the data to be visualized.
+        point_id (int): Point ID (used when sampling_method="SinglePoint").
+        sampling_method (str): Method for sampling data ("RandomPoint", "SinglePoint", or "Spatial").
+        fluid_domain_id (int or list): ID of the fluid domain
+        solid_domain_id (int or list): ID of the solid domain
+
+    Returns:
+        tuple: (Processed data type, DataFrame, Case name, Image folder, Hi-pass visualization folder).
+    """
+    folder_path = Path(folder)
+    case_name = folder_path.parent.name
+    visualization_path = folder_path / "Visualization"
+
+    logging.info(f"--- Processing folder path {folder_path}\n")
+
+    mesh_name_suffix = "" if save_deg == 1 else "_refined"
+    mesh_path = Path(mesh_path)
+    mesh_path = mesh_path.with_name(f"{mesh_path.stem}{mesh_name_suffix}{mesh_path.suffix}")
+    mesh_path_fluid = mesh_path.with_name(f"{mesh_path.stem}_fluid.h5")  # Needed for formatting SPI data
+
+    formatted_data_folder_name = f"res_{case_name}_stride_{stride}t{start_t}_to_{end_t}save_deg_{save_deg}"
```

Comment:
This formatted data folder name is hard to read. I think we can remove `case_name` because it is clear what folder we are working on and maybe can keep other parameters. 

So for example, 
```suggestion
    formatted_data_folder_name = f”npz_{start_t}s_to_{end_t}s_stride_{stride}save_deg_{save_deg}"
```
Other possibility is to make folder name more general and just add json file containing what parameters were used to generate these npz files, but that can be too much to do.